src/sentry/integrations/aws_lambda/client.py

```diff
@@ -70,3 +94,101 @@ def gen_aws_client(account_number, region, aws_external_id, service_name="lambda
+        aws_external_id: str,
+    ) -> None:
+        self.base_url = get_proxy_url()
+        self.account_number = account_number
+        self.region = region
+        self.aws_external_id = aws_external_id
+        super().__init__(org_integration_id=org_integration_id)
+
+    @property
+    def client(self):
+        if self._client:
+            return self._client
+
+        self._client = gen_aws_client(
+            account_number=self.account_number,
+            region=self.region,
+            aws_external_id=self.aws_external_id,
+        )
+        return self._client
+
+    def should_delegate(self) -> bool:
+        return True
+
+    def delegate(self, request, proxy_path: str, headers) -> HttpResponse:
+        payload = request.data
+        function_name = payload["function_name"]
+        args = payload["args"]
+        kwargs = payload["kwargs"]
+        boto3_func = getattr(self.client, function_name)
+        try:
+            result = boto3_func(*args, **kwargs)
+            return JsonResponse(
+                data={
+                    "function_name": function_name,
+                    "return_response": result,
+                    "exception": None,
+                },
+                status=200,
+            )
+        except Exception as err:
+            logger.info("boto3.client.exception", extra={"error": err})
+            return JsonResponse(
+                data={
+                    "function_name": function_name,
+                    "return_response": {},
+                    "exception": {"class": err.__class__.__name__},
+                },
+                status=400,
+            )
+
+    def __getattr__(self, func_name: str):
+        if SiloMode.get_current_mode() != SiloMode.REGION:
+
+            def boto3_func(*args, **kwargs):
+                func = getattr(self.client, func_name)
+                return func(*args, **kwargs)
+
+            return boto3_func
+
+        def boto3_proxy_func(*args, **kwargs):
+            # From the region silo, we create a request payload to the internal integration proxy endpoint.
+            payload = {
+                "args": list(args),
+                "kwargs": kwargs,
+                "function_name": func_name,
+            }
+
+            response = self.post("/", data=payload)
+            proxy_response = ProxyResponseSerializer(data=response)
+            if not proxy_response.is_valid():
+                raise Exception(f"Invalid response from calling: {func_name}")
+            validated_data = proxy_response.validated_data
+
+            function_name = validated_data["function_name"]
+            assert function_name == func_name
+
+            exception = validated_data["exception"]
+            if exception:
+                class_name = exception["class"]
+                lambda_client = self.client
+                exception_cls = getattr(lambda_client.exceptions, class_name)
+                raise exception_cls()
```

Comment:
I'm guessing this is to maintain parity with how boto3 works?

Reply:
To maintain parity of how the AWS Lambda uses the boto3 client.

Reply:
👍